src/main/resources/templates/db_scripts/budget_encumbrances_rollover.sql

```diff
@@ -422,53 +433,65 @@ CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.budget_encumbrances_rollo
+                            'metadata', ${myuniversity}_${mymodule}.budget.jsonb->'metadata' || jsonb_build_object('createdDate', date_trunc('milliseconds', clock_timestamp())::text));
+        END IF;
+
+        IF _rollover_record->>'rolloverType' <> 'Preview' THEN
+            -- #1.1 Create budget expense class relations for new budgets
+            INSERT INTO ${myuniversity}_${mymodule}.budget_expense_class
+            SELECT public.uuid_generate_v5(public.uuid_nil(), concat('BER6', oldBudget.id, fund.id, newBudget.id, exp.id)),
+                   jsonb_build_object('budgetId', newBudget.id,
+                                      'expenseClassId', exp.jsonb->>'expenseClassId',
+                                      'status', exp.jsonb->>'status')
+            FROM ${myuniversity}_${mymodule}.budget AS oldBudget
+                     INNER JOIN ${myuniversity}_${mymodule}.fund AS fund ON fund.id = oldBudget.fundId
+                     INNER JOIN ${myuniversity}_${mymodule}.budget AS newBudget ON newBudget.fundId = oldBudget.fundId
+                     INNER JOIN ${myuniversity}_${mymodule}.budget_expense_class AS exp ON oldBudget.id = exp.budgetid
+            WHERE oldBudget.jsonb ->> 'fiscalYearId' = _rollover_record->>'fromFiscalYearId'
+              AND fund.jsonb ->> 'ledgerId' = _rollover_record->>'ledgerId'
+              AND newBudget.jsonb->>'fiscalYearId' = _rollover_record->>'toFiscalYearId'
+            ON CONFLICT DO NOTHING;
+
+            -- #1.2 Create budget groups relation for new budgets
+            INSERT INTO ${myuniversity}_${mymodule}.group_fund_fiscal_year
+            SELECT public.uuid_generate_v5(public.uuid_nil(), concat('BER7', oldBudget.id, fund.id, newBudget.id, gr.id)),
+                   jsonb_build_object('budgetId', newBudget.id,
+                                      'groupId', gr.jsonb->>'groupId',
+                                      'fiscalYearId', _rollover_record->>'toFiscalYearId',
+                                      'fundId', gr.jsonb->>'fundId')
+            FROM ${myuniversity}_${mymodule}.budget AS oldBudget
+                     INNER JOIN ${myuniversity}_${mymodule}.fund AS fund ON fund.id = oldBudget.fundId
+                     INNER JOIN ${myuniversity}_${mymodule}.budget AS newBudget ON newBudget.fundId = oldBudget.fundId
+                     INNER JOIN ${myuniversity}_${mymodule}.group_fund_fiscal_year AS gr ON oldBudget.id = gr.budgetid
+            WHERE oldBudget.jsonb ->> 'fiscalYearId' = _rollover_record->>'fromFiscalYearId'
+              AND fund.jsonb ->> 'ledgerId' = _rollover_record->>'ledgerId'
+              AND newBudget.jsonb->>'fiscalYearId' = _rollover_record->>'toFiscalYearId'
+            ON CONFLICT DO NOTHING;
+        END IF;
+
+        CREATE TEMPORARY TABLE tmp_transaction(LIKE ${myuniversity}_${mymodule}.transaction);
```

Comment:
i think you should add IF NOT EXISTS or similar to this, just in case.

Reply:
budget_encumbrance_rollover function - it basically starting point of execution, and temporary table creating per transaction, so we should not meet situation when this temp table already exists

Reply:
I see. Thanks!